Source/src_android/PinCheckerLibrary/src/main/java/com/wultra/android/passphrasemeter/PasswordTester.java

```diff
@@ -150,13 +158,30 @@ public PinTestResult testPin(@NonNull String pin) throws WrongPinException {
             }
         }
 
-        return new PinTestResult(pin, set);
+        return new PinTestResult(pinBytes.length, set);
     }
 
     // Private methods & constants
 
     /**
-     * Defines constants returned from {@link #testPasswordJNI(String)} method.
+     * Converts user's byte input into ByteBuffer
+     *
+     * @param byteArray Password or PIN as byte array
+     * @param arrayLength length of an array
+     * @return Shared byte buffer between Java/JNI
+     */
+    private ByteBuffer getSharedBuffer(byte[] byteArray, int arrayLength) {
+        ByteBuffer bb = ByteBuffer.allocateDirect(arrayLength);
+        bb.put(byteArray);
+        // This step might be too much but it is better than to just set this reference to null and
+        // hope for "quick" GC
+        Arrays.fill(byteArray, (byte) 0);
```

Comment:
We don't own the byte array, so library should not modify its content. As a library developers we don't know what application's doing afterwards with its password stored in the array. I would keep it as it is and note this to the documentation that application should properly cleanup the password stored in the array.

Look at [Password.validatePasswordComplexity()](https://github.com/wultra/powerauth-mobile-sdk/blob/develop/proj-android/PowerAuthLibrary/src/main/java/io/getlime/security/powerauth/core/Password.java#L271). That's the actual point where PA-SDK meets with the password meter.